### ISSUE TITLE
ClassNotFoundException: io.flutter.actions.FlutterNewProjectAction PluginClassLoader

### ISSUE THREAD

**Issue by BorzdeG** [flutter/flutter-intellij#2582]
## What happened

(please describe what you were doing when this exception occurred)

## Version information

Android Studio `3.3 Canary 8` • Flutter plugin `io.flutter 27.1.3` • Dart plugin `182.3911.37`

Flutter 0.7.6-pre.20 • channel master • git@github.com:BorzdeG/flutter.git
Framework • revision 457e7b23df (2 days ago) • 2018-09-01 04:06:46 -0700
Engine • revision 017dafd426
Tools • Dart 2.1.0-dev.3.0.flutter-760a9690c2

## Exception

null

```
com.intellij.diagnostic.PluginException: io.flutter.actions.FlutterNewProjectAction PluginClassLoader[io.flutter, 27.1.3] com.intellij.ide.plugins.cl.PluginClassLoader@3d43e2e7 [Plugin: io.flutter]
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:147)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getActionImpl(ActionManagerImpl.java:461)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getAction(ActionManagerImpl.java:449)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.preloadActions(ActionManagerImpl.java:1281)
	at com.intellij.openapi.actionSystem.impl.ActionPreloader.preload(ActionPreloader.java:32)
	at com.intellij.openapi.application.Preloader.lambda$null$0(Preloader.java:74)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.application.Preloader.lambda$initComponent$1(Preloader.java:72)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:226)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2$1.run(BoundedTaskExecutor.java:199)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:229)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:193)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: io.flutter.actions.FlutterNewProjectAction PluginClassLoader[io.flutter, 27.1.3] com.intellij.ide.plugins.cl.PluginClassLoader@3d43e2e7
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:63)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:143)
	... 19 more
```

**Comment by jwren** [comment]
Assigning to Steve as this is his wheel-house.

**Comment by stevemessick** [comment]
Canary builds are not supported.

**Comment by stevemessick** [comment]
FWIW I tracked this down to what I believe is a build issue in Android Studio. It apparently does not define the module used to define dependencies for plugins. (The modules that identify different products are documented by Jetbrains.) The problem is also present in Canary 10. I'm sure it will eventually get fixed; I saw the same problem in the 3.0 beta last year. But for now, you can't use the latest. I think Canary 7 was OK.
